boot/mbed/app_enc_keys.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020 Embedded Planet
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+#include <bootutil/sign_key.h>
+#include <mcuboot_config/mcuboot_config.h>
+
+#if defined(MCUBOOT_SIGN_RSA)
+#define HAVE_KEYS
+extern const unsigned char rsa_pub_key[];
+extern unsigned int rsa_pub_key_len;
+#elif defined(MCUBOOT_SIGN_EC256)
+#define HAVE_KEYS
+extern const unsigned char ecdsa_pub_key[];
+extern unsigned int ecdsa_pub_key_len;
+#elif defined(MCUBOOT_SIGN_ED25519)
+#define HAVE_KEYS
+extern const unsigned char ed25519_pub_key[];
+extern unsigned int ed25519_pub_key_len;
+#else
+#error "No public key available for given signing algorithm."
+#endif
+
+/*
+ * NOTE: *_pub_key and *_pub_key_len are autogenerated based on the provided
+ *       key file. If no key file was configured, the array and length must be
+ *       provided and added to the build manually.
```

Comment:
Who is generating the key files for this port?

Reply:
Is there a way to disable encryption for this PR at least? I haven't tested it yet.

At some point I believe I had made changes to allow the application encryption keys to be defined by the user. They would then use the imgtool tools to generate .C versions of their encryption keys. Would this be an acceptable path forward?

Reply:
You don't need encryption if you decide so, but add some #error message if the user has enabled it.

Reply:
Oh, I didn't read the whole file. The key symbols are declared as externs and so can be provided by the user if they wish to enable encryption.

Can we just reword the comment? The encryption keys can be generated using imgtool the same way the signing keys are generated. There is no automated build step provided by Mbed-OS to do this.

Reply:
Sure, using `imgtool getpub ...`  should be fine!